### ISSUE TITLE
Server Crash with Automated Digital Miner (Ticking Block Entity)

### ISSUE THREAD

**Issue by C4TSCR47CH** [mekanism/Mekanism#6258]
*Please use the search functionality before reporting an issue. Also take a look at the closed issues!*

#### Issue description:
I host a server of a custom modpack, that I've ended up building an automated digital miner in. It works off of OpenComputers, ProjectRed frames, and, of course, the digital miner. I'm having an issue where the server will crash while the miner is active, with a "Ticking block entity" error. After reading the trace, it seems to be localized in Mekanism. Before posting this, I've had this crash three times, each of which I've tried to investigate the issue, and came up with no results. The miner is built in Aroma1997's Dimensional World, if that makes any difference.

#### Steps to reproduce:
Unknown if this is reproducable, as there is a lot that I've done to make this miner work. But, I'll give my code for my open computers miner and a video showcasing the miner's build. I'm aware the code isn't great, as I'm fairly new to Lua and this pack is designed all around learning mods.
1. Build a digital miner automation system (see video)
2. Run miner for a while
3. Potentially crash? Still unknown if this is an issue from the miner itself or what.

#### Version (make sure you are on the latest version before reporting):
**Forge:** 14.23.5.2854 (MC 1.12.2)
**Mekanism:** 9.8.3.390
**Other relevant version:** 

#### If a (crash)log is relevant for this issue, link it here: (_It's almost always relevant_)

[pastebin](https://pastebin.com/XE3TuJff)

Code for digital miner automation:

[pastebin](https://pastebin.com/CQvQfKrT)

Video showcasing miner (timestamps in description):

[youtube](https://www.youtube.com/watch?v=9V06cQGTqD8&feature=youtu.be)

**Comment by C4TSCR47CH** [comment]
I'm going to continue testing stuff, but I have both a normal chunkloader on the frame machine, as well as an anchor upgrade in the miner, if that matters at all. All upgrades in the miner are full speed upgrades, energy upgrades, and that anchor upgrade. Based on the other issue listed about the miner crashing in the end, I'm going to try to remove the anchor upgrade and see if that fixes the problem.

**Comment by thiakil** [comment]
I really can't see what would be modifying the list from outside....

**Comment by C4TSCR47CH** [comment]
Well, all was looking good for a while, but just now, it crashed again. This time, I was in the overworld, away from the miner, and I got an error about the overworld not being loaded, when it kicked me out.

Here's the crash report, pretty much the same as before, as far as I can tell.
[pastebin](https://pastebin.com/LAwSYgfx)

As well as the full log, that includes the whole unloading overworld part. (There's also some stuff to do with Applied Energistics Autocrafting not recognizing energy cubes in patterns, as I'm trying to automate making a full size induction multiblock)
[pastebin](https://pastebin.com/Qn8T7ES4)

**Comment by C4TSCR47CH** [comment]
Could the issue be coming from weird timing in the Lua program, where it tries to do two things at once potentially? One that Mekanism itself tells the miner to do, and another where the program tells it to do something? I'm sorry if it sounds dumb, I don't know exactly what's going on, but based on a bit of research on the Concurrent Modification Exception error, this is the only thing I can expect to be happening. After all, my code isn't quite perfect, by any means. Lua is still really foreign to me, and just getting this thing to work was kind of a miracle.

In the meantime, I've changed my code a bit in another testing program, to see if the issue comes back up after a while. Due to it happening an hour or two after the miner is active, it's kinda annoying having to wait, but hey, I can do other things in the meantime while it runs.

**Comment by thiakil** [comment]
It appears to be when it's running out of power, based on the stack trace

**Comment by C4TSCR47CH** [comment]
See, that shouldn't be it either. The quantum entangloporter is feeding the miner with power from a fusion reactor producing at 400k rf/t, and even with full speed and full energy, it doesn't seem to be emptying that porter at all. Not to mention, the block data given by the crash report states that the miner has completely full energy.

The new program I wrote hasn't caused a crash since I last posted, but the problem comes from now there's no way to check based on the actual miner updating, I'm just estimating it based on full speed mining. If I removed speed upgrades, the program would be inaccurate. It's fine for my use case, honestly, so I really don't care as long as my server stops crashing. So far, it's been successful, so I'm tempted to say it was something up with my programming, which honestly wouldn't surprise me.
